crates/project/src/debugger/locators/go.rs

```diff
@@ -0,0 +1,322 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use collections::FxHashMap;
+use dap::{DapLocator, DebugRequest, adapters::DebugAdapterName};
+use gpui::SharedString;
+use std::path::PathBuf;
+use task::{DebugScenario, SpawnInTerminal, TaskTemplate};
+
+pub(crate) struct GoLocator;
+
+#[async_trait]
+impl DapLocator for GoLocator {
+    fn name(&self) -> SharedString {
+        SharedString::new_static("go-debug-locator")
+    }
+
+    fn create_scenario(
+        &self,
+        build_config: &TaskTemplate,
+        resolved_label: &str,
+        adapter: DebugAdapterName,
+    ) -> Option<DebugScenario> {
+        if build_config.command != "go" {
+            return None;
+        }
+
+        let go_action = build_config.args.first()?;
+
+        match go_action.as_ref() {
+            "run" => {
+                let program = build_config
+                    .args
+                    .get(1)
+                    .cloned()
+                    .unwrap_or_else(|| ".".to_string());
+                let args = build_config.args.get(2..).unwrap_or(&[]).to_vec();
+
+                let config = serde_json::json!({
+                    "request": "launch",
+                    "mode": "debug",
+                    "program": program,
+                    "args": args,
+                    "cwd": build_config.cwd.as_deref().unwrap_or("${ZED_WORKTREE_ROOT}"),
+                    "env": build_config.env,
+                    "console": "integratedTerminal",
+                    "outputMode": "remote"
+                });
+
+                Some(DebugScenario {
+                    label: resolved_label.to_string().into(),
+                    adapter: adapter.0,
+                    build: None, // No build task - Delve handles everything
```

Comment:
One problem with this is that we have an arbitrary timeout of 12 seconds on the initialize request, which is (IIRC) when delve runs the build. Thus it's possible for startup sequence to time out when done this way. :/